### PR TITLE
8311033: [macos] PrinterJob does not take into account Sides attribute

### PR DESCRIPTION
**Pull request by AlexanderScherbatiy**
To reproduce the issue run the [JavaSidesAttributePrinting](https://bugs.openjdk.org/secure/attachment/104448/JavaSidesAttributePrinting.java) java sample with ONE_SIDED and DUPLEX arguments on macOS:
```
java JavaSidesAttributePrinting ONE_SIDED
java JavaSidesAttributePrinting DUPLEX
```

The sample calls PrinterJob print method with the given sides attribute.
The pages are printed according to the printer default settings not according to the provided Sides attributes.

The fix propagates Sides attribute
- from `PrinterJob` to `NSPrintInfo` in `CPrinterJob.javaPrinterJobToNSPrintInfo` method
- from `NSPrintInfo` to `PrinterJob` in `CPrinterJob.nsPrintInfoToJavaPrinterJob` method

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311033](https://bugs.openjdk.org/browse/JDK-8311033): [macos] PrinterJob does not take into account Sides attribute (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [4480595c](https://git.openjdk.org/jdk/pull/14727/files/4480595c037197264015f16191037fbdb9d716db)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14727/head:pull/14727` \
`$ git checkout pull/14727`

Update a local copy of the PR: \
`$ git checkout pull/14727` \
`$ git pull https://git.openjdk.org/jdk.git pull/14727/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14727`

View PR using the GUI difftool: \
`$ git pr show -t 14727`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14727.diff">https://git.openjdk.org/jdk/pull/14727.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14727#issuecomment-1614437303)

### CODE REVIEW

**Comment by prrace** [comment]
1) The test doesn't seem to check whether the printer actually supports two-sided printing (!)
 You need to see if the value you use is supported else skip it

2) The test text is so similar .. and the user clearly is expected to "know" what "TWO_SIDED_LONG_EDGE" means.
I think you need to add a window that explains that "up to" THREE tests will run and it will test all the cases supported by the printer and you need to add words explaining each case

3) SHORT_EDGE doesn't seem to work for me. Hmm not working from Safari either so that can't be JDK code's problem

**Comment by AlexanderScherbatiy** [comment]
The test is updated to check the supported sides attributes. The test automatically passes if the number of supported attributes is equal or less than one.
A simple explanation is added which describes what needs to be printed when one sided, duplex, or tumble attribute is used.

**Review by prrace — APPROVED** [review]

File: src/java.desktop/macosx/native/libawt_lwawt/awt/CPrinterJob.m [hunk 1]
```diff
@@ -496,6 +527,17 @@ static void javaPrinterJobToNSPrintInfo(JNIEnv* env, jobject srcPrinterJob, jobj
     } else {
        [dst setJobDisposition:NSPrintSpoolJob];
     }
+
+    int sides = (*env)->CallIntMethod(env, srcPrinterJob, jm_getSides);
```
Comment (mrserb) [inline src/java.desktop/macosx/native/libawt_lwawt/awt/CPrinterJob.m:531]: it seems the code around uses the "java" types like jint.
>jint copies = (*env)->CallIntMethod(env, srcPrinterJob, jm_getCopies);

Comment (AlexanderScherbatiy) [reply src/java.desktop/macosx/native/libawt_lwawt/awt/CPrinterJob.m]: Could you review the updated fix?

`int` type is changed to `jint` for sides variable.

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**